### PR TITLE
Fix Region selection bug

### PR DESCRIPTION
**Pull request by Jujuyeh**
- Fixed region selection that wouldn't work as intended for US West.
- Changed error messages to better explain the issue.

### CODE REVIEW

File: common/prepare-track/init.sh [hunk 1]
```diff
@@ -296,11 +297,7 @@ function deploy_agent () {
 # Test if the Agent connected successfully to the collector endpoint.
 ##
 function test_agent () {
-    if [ "$USE_MONITOR_API" == true ] || [ "$USE_SECURE_API" == true ]; then
```
Comment (pmusa) [inline common/prepare-track/init.sh:299]: Why did you change this?

File: common/prepare-track/init.sh [hunk 2]
```diff
@@ -309,7 +306,7 @@ function test_agent () {
     while [ "$connected" != true ] && [ $attempt -le $MAX_ATTEMPTS ]
     do
         sleep 3
-        kubectl logs -l app.kubernetes.io/instance=sysdig-agent -n sysdig-agent --tail=-1 2> /dev/null | grep "${AGENT_COLLECTOR}" &> /dev/null
+        kubectl logs -l app.kubernetes.io/instance=sysdig-agent -n sysdig-agent --tail=-1 2> /dev/null | grep "Connected to collector" &> /dev/null
```
Comment (pmusa) [inline common/prepare-track/init.sh:309]: Why did you change this?

File: common/prepare-track/init.sh [hunk 3]
```diff
@@ -322,10 +319,21 @@ function test_agent () {
 
     if [ "$connected" = true ]
     then
-        echo "  OK"
-        touch $WORK_DIR/user_data_AGENT_OK
+        FOUND_COLLECTOR=`kubectl logs -l app.kubernetes.io/instance=sysdig-agent -n sysdig-agent --tail=-1 2> /dev/null | grep "collector:" | head -n1 | rev | cut -d' ' -f 1 | rev`
+        if [ "${FOUND_COLLECTOR}" == "${AGENT_COLLECTOR}"]
```
Comment (pmusa) [inline common/prepare-track/init.sh:323]: I am not sure I follow the logic here. Can you elaborate on it?

**Review by pmusa — COMMENTED** [review]
I am not sure I understand some of the changes. i commented on the code chunks

File: common/prepare-track/init.sh (hunk 2 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:309]: We cannot ensure the agent works until it's connected to the collector. Agent logs which collector is going to try to connect to, it outputs its settings so it's not guaranteed that the agent will connect. A check that tells if we are using the right connector is used later.

File: common/prepare-track/init.sh (hunk 3 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:323]: We will consider the Agent is working when:
- It's connected to the collector
- It's set to the right collector

File: common/prepare-track/init.sh (hunk 1 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:299]: It was a condition that just added indentation to the output of a message.

File: common/prepare-track/init.sh (hunk 2 above)
Comment (pmusa) [reply common/prepare-track/init.sh:309]: The agent logs both strings regardless of a successful connection. So, both tests are not testing if the connection was successful. After some investigation, I found a message that is logged after a successful connection:
```
2021-12-14 11:21:10.293, 18857.18907, Information, connection_manager:1103: Sending scraper version promscrape_v2 to backend
```

I think the agent also logs:
```
2021-12-14 11:21:10.456, 18857.18907, Information, async_aggregator:85: timestamp_ns: 1639480870435000000
machine_id: "42:01:0a:84:01:8a"
customer_id: "baeb7dda-4576-4fac-b4f8-d830fadfc18a"
```

Comment (Jujuyeh) [reply common/prepare-track/init.sh:309]: Agent logs consists in:

```
- Here are my settings
- Bootstrapping
- Connected to collector / Error if can't connect
- Liveness messages, connections, etc.
```

Comment (pmusa) [reply common/prepare-track/init.sh:309]: This is the log for an error. It also has "connected to collector".
```
2021-12-14 10:53:55.321, 7255.7342, Information, connection_manager:446: Connected to collector
2021-12-14 10:53:55.321, 7255.7307, Information, connection_manager:714: Performing protocol handshake
2021-12-14 10:53:55.321, 7255.7307, Information, connection_manager:1664: Sent msgtype=24 len=158 to collector
2021-12-14 10:53:55.321, 7255.7307, Information, connection_manager:1281: Waiting for 60 seconds for handshake to complete
2021-12-14 10:53:55.486, 7255.7307, Error, connection_manager:2185: Received error message: ERR_INVALID_CUSTOMER_KEY (Unauthorized agent access key)
2021-12-14 10:53:55.486, 7255.7307, Information, connection_manager:566: Disconnecting from collector
2021-12-14 10:53:55.486, 7255.7307, Information, connection_manager:349: Initiating connection to collector
```

File: common/prepare-track/init.sh (hunk 3 above)
Comment (pmusa) [reply common/prepare-track/init.sh:323]: I don't see how it can be connected to the wrong collector and be connected. But if you think that this is a valid case, ok.

File: common/prepare-track/init.sh (hunk 1 above)
Comment (pmusa) [reply common/prepare-track/init.sh:299]: Yes, that is because of the way we present the messages. When monitor and secure are active, we run the agent in the background and keep running the script. So, the agent message is better of without indentation.

File: common/prepare-track/init.sh (hunk 2 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:309]: OK, this happens when either the Agent key is wrong or the user selected the wrong region. A third check can be added for this.

Comment (Jujuyeh) [reply common/prepare-track/init.sh:309]: Sorry just saw the other message, I added a second comment without refreshing

Comment (Jujuyeh) [reply common/prepare-track/init.sh:309]: when can replace Connected to collector for a message that we know always will be logged after a successful connection (correct region and keys)

```
Sending scraper version promscrape_v2 to backend
```

 Might do the trick, but not sure if promscrape is something only enabled in prometheus scenarios

File: common/prepare-track/init.sh (hunk 1 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:299]: OK I see your point. Multiple jobs sharing a an output is always messy. One common solution is to add the name of the job and its output, so that it can be followed as different logs:

```
[agent]          Yada yada.
[api-endpoint]   Example example.
[agent]          Yada yada part 2.
```

 but an overkill for now...

File: common/prepare-track/init.sh (hunk 2 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:309]: I'll try to reach out to an SME to get the best solutions and maybe a better explanation on the agent logs

File: common/prepare-track/init.sh (hunk 3 above)
Comment (pmusa) [reply common/prepare-track/init.sh:323]: sounds good to me.

File: common/prepare-track/init.sh (hunk 1 above)
Comment (pmusa) [reply common/prepare-track/init.sh:299]: I vote to keep the `if` and the indentation. looks cleaner.

File: common/prepare-track/init.sh (hunk 2 above)
Comment (pmusa) [reply common/prepare-track/init.sh:309]: For now, I would change it to the scraper message and update it later if we get a good answer from the SME.

File: common/prepare-track/init.sh (hunk 1 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:299]: OK, I'll bring it back :+1:

File: common/prepare-track/init.sh (hunk 2 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:309]: OK, I'll solve this as you said, and I'll open a new issue to get various future improvements in the init.sh script when we have SME feedback :+1:

File: common/prepare-track/init.sh [hunk 4]
```diff
@@ -296,20 +297,22 @@ function deploy_agent () {
...
 ##
 function test_agent () {
-    if [ "$USE_MONITOR_API" == true ] || [ "$USE_SECURE_API" == true ]; then
-      echo "Testing if Sysdig Agent is running correctly..."
+    if [ "$USE_MONITOR_API" == true ] || [ "$USE_SECURE_API" == true ]
+    then
+        echo "Testing if Sysdig Agent is running correctly..."
     else
-      echo "  Testing if Sysdig Agent is running correctly..."
+        echo -n "  Testing if Sysdig Agent is running correctly..."
```
Comment (pmusa) [inline common/prepare-track/init.sh:304]: Are you sure we want a `-n` in this case? I dont remember feeling the need for it. Actually, it would be nice to have a line break after it.

File: common/prepare-track/init.sh [hunk 5]
```diff
@@ -322,10 +325,21 @@ function test_agent () {
 
     if [ "$connected" = true ]
     then
-        echo "  OK"
-        touch $WORK_DIR/user_data_AGENT_OK
+        FOUND_COLLECTOR=`kubectl logs -l app.kubernetes.io/instance=sysdig-agent -n sysdig-agent --tail=-1 2> /dev/null | grep "collector:" | head -n1 | rev | cut -d' ' -f 1 | rev`
+        if [ "${FOUND_COLLECTOR}" == "${AGENT_COLLECTOR}"]
+        then
+            echo "  OK"
```
Comment (pmusa) [inline common/prepare-track/init.sh:331]:
```suggestion
            echo "  Sysdig Agent successfully installed."
```

**Review by pmusa — CHANGES_REQUESTED** [review]

File: common/prepare-track/init.sh (hunk 4 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:304]: When only deploying the Agent it would be:

```
  Testing if Sysdig Agent is running correctly...  OK
```

File: common/prepare-track/init.sh (hunk 5 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:331]: That is another approach

Comment (Jujuyeh) [reply common/prepare-track/init.sh:331]: Going for your approach, as it would be more clear to the user when running the agent on the background

File: common/prepare-track/init.sh (hunk 4 above)
Comment (Jujuyeh) [reply common/prepare-track/init.sh:304]:
```suggestion
        echo "  Testing if Sysdig Agent is running correctly..."
```

File: common/prepare-track/init.sh [hunk 6]
```diff
@@ -296,20 +297,22 @@ function deploy_agent () {
...
+    then
+        echo "Testing if Sysdig Agent is running correctly..."
     else
-      echo "  Testing if Sysdig Agent is running correctly..."
+        echo "  Testing if Sysdig Agent is running correctly..."
     fi
 
     attempt=0
     MAX_ATTEMPTS=40 # 2 minutes
+    CONNECTED_MSG="Sending scraper version promscrape_v2 to backend"
```
Comment (pmusa) [inline common/prepare-track/init.sh:309]: Is this not going to break when v2 becomes something else?

Comment (pmusa) [reply common/prepare-track/init.sh:309]:
```suggestion
    CONNECTED_MSG="Sending scraper version"
```

**Review by pmusa — APPROVED** [review]
LGTM after this last change.
